tests/bluetooth/tester/src/audio/btp_bap_broadcast.h

```diff
@@ -41,6 +42,8 @@ struct btp_bap_broadcast_remote_source {
 };
 
 struct btp_bap_broadcast_local_source {
+	bool allocated;
+	uint8_t source_id;
```

Comment:
If this value is the same as the `ext_adv_idx`, suggest to use that here as well
```suggestion
	uint8_t ext_adv_idx;
```

Assuming that they are in the same value space. 

Alternatively we need to use both a `source_id` and a `ext_adv_idx`. Consider the scenario where we need 2 advertising sets:
Connectable advertising set using `ext_adv_idx=0`
Non-connectable (broadcast) advertising sets using `source_id=0` but also `ext_adv_idx=1`. 

So perhaps we should not treat `source_id == ext_adv_idx` the same (one is also an ID, and the other is an index). 
There's also the option that `tester_gap_create_adv_instance` should not _take_ an index/SID, but rather _return_ an index/SID or perhaps even a pointer to a `struct bt_le_ext_adv`. 

What we really need here it to be able to create an extended advertising set from a `btp_bap_broadcast_local_source` and keep a reference to that, right? Which can be done with an ID, index or pointer. 

Only if we really need to be able to control the SID do we need to supply that when creating the set.

Reply:
We do need to make sure the SIDs are different for this specific test though

Reply:
The call to `tester_gap_create_adv_instance` can be responsible for assigning unique SIDs

Reply:
Done